Updates for injection of pluggable components. (#3514)
dapr/docs: daprdocs/content/en/operations/components/pluggable-components-registration.md

Both your component and the Unix Socket must be running before Dapr starts.

By default, Dapr sidecar looks for components as [Unix Domain Socket][uds] files in `/tmp/dapr-components-sockets`.

Filenames in this folder are significant for component registration. They must be formed by appending the component's **name** with a file extension of your choice, more commonly `.sock`. For example, the filename `my-component.sock` is a valid Unix Domain Socket file name for a component named `my-component`.

> Since you are running Dapr in the same host as the component, verify this folder and the files within it are accessible and writable by both your component and Dapr. If you are using Dapr's sidecar injector capability, this volume is created and mounted automatically.

Reviewer: ```suggestion
Since you are running Dapr in the same host as the component, verify that this folder and the files within it are accessible and writable by both your component and Dapr. If you are using Dapr's sidecar injector capability, this volume is created and mounted automatically.
```

[Initialize Dapr]({{< ref get-started-api.md >}}), and make sure that your component file is placed in the right folder.

{{% alert title="Note" color="primary" %}}

> Dapr 1.9.0 is the minimum version that supports pluggable components. While version 1.11 supports automatically injection of the containers for pluggable components.

Reviewer: ```suggestion
Dapr 1.9.0 is the minimum version that supports pluggable components. As of version 1.11.0, automatic injection of the containers is supported for pluggable components.
```

kind: Component
metadata:
  name: prod-mystore
  # When running on Kubernetes and automatic container injection, add annotation below:
  annotations:
    dapr.io/component-container: >
      {
        "name": "my-component",
        "image": "<registry>/<image_name>:<image_tag>"
      }
spec:
  type: your_component_type.your_socket_goes_here
  version: v1
  metadata:
scopes:
  - backend
```

> * The `dapr.io/component-container` annotation is mandatory when running on Kubernetes and you want Dapr's sidecar injector to handle the container and volume injection for the pluggable component. The `name` and `image` attributes are the minimum for the Dapr's sidecar injector to successfully add the container to the Application's pod. Volume for Unix Domain Socket is automatically created and mounted by Dapr's sidecar injector.

Reviewer: ```suggestion
The `dapr.io/component-container` annotation is mandatory on Kubernetes when you want Dapr's sidecar injector to handle the container and volume injection for the pluggable component. At minimum, you'll need the `name` and `image` attributes for the Dapr's sidecar injector to successfully add the container to the Application's pod. Volume for Unix Domain Socket is automatically created and mounted by Dapr's sidecar injector.
```
